Source/Game/Game/Editor/EaseCurveTool.cpp

```diff
@@ -0,0 +1,402 @@
+            ImGui::SameLine();
+            ImGui::PushItemWidth(160);
+            if (ImGui::BeginCombo("##EaseComboBoxName", easeName[selectedEaseName]))
+            {
+                for (i32 i = 0; i < 14; i++)
+                {
+                    bool isSelected = (selectedEaseName == i);
+                    if (ImGui::Selectable(easeName[i], isSelected))
+                    {
+                        selectedEaseName = i;
+                        ResetTimer();
+                        ResetMinMax();
+                    }
+
+                    if (isSelected)
+                        ImGui::SetItemDefaultFocus();
+                }
+                ImGui::EndCombo();
+            }
+            ImGui::PopItemWidth();
+
+            if (selectedEaseName == 13)
+            {
+                ImGui::SeparatorText("Parameters");
+                ImGui::PushItemWidth(100);
+                ImGui::DragInt("Polynomial Degree", &polynomialDegree, 1, 2, 32);
+                ImGui::PopItemWidth();
+            }
+
+            ImGui::SeparatorText("Animation Timer");
+            ImGui::PushItemWidth(160);
+            if (ImGui::DragFloat("seconds##EaseTimer", &_timerScale, 0.001f, 1.f, 5.f, "%.3f", ImGuiSliderFlags_ClampOnInput))
+            {
+                ResetTimer();
+                _inverse = false;
+            }
+            ImGui::PopItemWidth();
+
+            ImGui::SeparatorText("2D Preview");
+
+            ImGui::PushItemWidth(100);
+            if (ImGui::DragFloat("Canvas Scale", &canvasScale, 0.01f, 0.5f, 0.8f))
+                invHalfCanvasScale = (1.f - canvasScale) / 2.f;
+            ImGui::PopItemWidth();
+
+            ImGui::PushItemWidth(100);
+            ImGui::DragInt("Number of segments", &segment, 1.f, 5, 50);
+            ImGui::PopItemWidth();
+
+            if (ImGui::IsItemHovered())
+            {
+                ImGui::SetTooltip("This is only for curve preview");
+            }
+
+            ImGui::Separator();
+
+            if (_preview2D)
+            {
+                ImVec2 size(320.f, 320.f);
+                ImGui::InvisibleButton("EaseCanvas", size);
+                ImVec2 p0 = ImGui::GetItemRectMin();
+                ImVec2 p1 = ImGui::GetItemRectMax();
+                ImDrawList* drawList = ImGui::GetWindowDrawList();
+                drawList->PushClipRect(p0, p1);
+
+                ImVec2 VLTop = {p0.x + (size.x * invHalfCanvasScale), p0.y};
+                ImVec2 VLBottom = {VLTop.x, p1.y};
+                ImVec2 VRTop = {p1.x - (size.x * invHalfCanvasScale), p0.y};
+                ImVec2 VRBottom = {VRTop.x, p1.y};
+                ImVec2 HBLeft = {p0.x, p1.y - (size.y * invHalfCanvasScale)};
+                ImVec2 HBRight = {p1.x, HBLeft.y};
+                ImVec2 HTLeft = {p0.x, p0.y + (size.y * invHalfCanvasScale)};
+                ImVec2 HTRight = {p1.x, HTLeft.y};
+
+                drawList->AddLine(VLTop, VLBottom, IM_COL32(255, 255, 255, 100));
```

Comment:
Perhaps we could make the color value use in a variable (This occurs multiple times in the file, where you have the same color being constructured multiple times in a row, please fix this)